helix-core/src/main/java/org/apache/helix/task/WorkflowDispatcher.java

```diff
@@ -525,6 +525,8 @@ private void cleanupWorkflow(String workflow) {
         // clean all the contexts even if Configs and IdealStates are exists. Then all the workflows
         // and jobs will rescheduled again.
         removeContexts(workflow, jobs, _clusterDataCache.getTaskDataCache());
+        // Request for full-data refresh to re-fetch workflow resource configs.
+        _clusterDataCache.requireFullRefresh();
```

Comment:
Thanks @junkaixue for the review. 
So we need updated resourceConfig, so another thing we can do here is:
`_clusterDataCache.notifyDataChange(ChangeType.RESOURCE_CONFIG)` 

so that in the next pipeline run, whichever Cluster event type it is, the resourceConfig will always be read from the ZK. That way, we will only making full refresh of ResourceConfigs.

Reply:
Another thing we can do is, put the cacheRefresh/notifyDataChange logic only in the code-paths where the `TargetState` is set to `DELETE` i.e. when workflow is explicitly cancelled by the user. This use case is very rare than the standard workflow deletion i.e. workflow expiry.

Reply:
Updated the PR @junkaixue.